### ISSUE TITLE
Adding property to Array.prototype in page object file causes error

### ISSUE THREAD

**Issue by HorseJump** [cucumber/cucumber-js#1111]
I'm using this with protractor, but the files in the stack trace are all Cucumber.

Error:
Unhandled rejection TypeError: Cannot read property '0' of undefined
    at EventDataCollector.storePickle (C:\Testing\Automated-Browser-Testing\node_modules\cucumber\src\formatter\helpers\event_data_collector.js:48:30)
    at emitOne (events.js:116:13)
    at EventEmitter.emit (events.js:211:7)
    at C:\Testing\Automated-Browser-Testing\node_modules\cucumber\src\cli\helpers.js:59:26
    at Array.forEach (<anonymous>)
    at C:\Testing\Automated-Browser-Testing\node_modules\cucumber\src\cli\helpers.js:54:10
    at Generator.next (<anonymous>)
    at Generator.tryCatcher (C:\Testing\Automated-Browser-Testing\node_modules\bluebird\js\release\util.js:16:23)
    at PromiseSpawn._promiseFulfilled (C:\Testing\Automated-Browser-Testing\node_modules\bluebird\js\release\generators.js:97:49)
    at C:\Testing\Automated-Browser-Testing\node_modules\bluebird\js\release\generators.js:201:15
    at getTestCases (C:\Testing\Automated-Browser-Testing\node_modules\cucumber\lib\cli\helpers.js:102:18)
    at C:\Testing\Automated-Browser-Testing\node_modules\cucumber\src\cli\helpers.js:32:13
    at Generator.next (<anonymous>)
    at Generator.tryCatcher (C:\Testing\Automated-Browser-Testing\node_modules\bluebird\js\release\util.js:16:23)
    at PromiseSpawn._promiseFulfilled (C:\Testing\Automated-Browser-Testing\node_modules\bluebird\js\release\generators.js:97:49)
    at Promise._settlePromise (C:\Testing\Automated-Browser-Testing\node_modules\bluebird\js\release\promise.js:574:26)

When this code is removed, there is no issue. Also, if I add it in my world constructor there are no issues:
`
Array.prototype.allAreTrue = function(expected) {
 // do stuff
}
`

I did some debugging on my own and traced it back to here:
`
{
    key: 'storePickle',
    value: function storePickle(_ref4) {
      var pickle = _ref4.pickle,
          uri = _ref4.uri;

      this.pickleMap[uri + ':' + pickle.locations[0].line] = pickle;
    }
`

**Comment by charlierudolph** [comment]
I think that error might be in gherkin or somewhere down the tree. Do you need to extend the array prototype or can you get by without doing that? I don't think this is worth chasing down, but if you can find the issue, I'm happy to accept a PR / help determine a solution

**Comment by HorseJump** [comment]
Thank you for the reply. I can manage without it with just adding it to my custom world constructor. I did however notice that I wasn't able to add anything to Object's prototype either without a different error. Again, I found a workaround. I plan to contribute to the project in the near future and I'll see if I can narrow it down then when I have an idea of how the code works and I'm not looking at babelized code like that above.
